guides/plugins/apps/app-signing.md

```diff
@@ -0,0 +1,165 @@
+# App system signing
+
+To ensure secure communication between Shopware shops and your app server, Shopware signs all outgoing requests using a cryptographic signature.
+The signature is generated using HMAC-SHA256, hashing either the query string or the request body, depending on the request method, with your app secret.
+By verifying this signature on your server, you can confirm that the request originates from Shopware and remains unaltered during transmission.
+This mechanism safeguards your app against request forgery and unauthorized access.
+
+::: warning
+**Breaking Change Considerations**
+
+Shopware may change the parameters used for signature generation without considering it a breaking change.
+Your app should be flexible enough to handle variations in the signature generation data.
```

Comment:
I would just write about "adding". Removing is a breaking change of course and will only happen for a major.